controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -334,55 +374,84 @@ func (r *ReconcileHyperConverged) Reconcile(ctx context.Context, request reconci
 
 // resolveReconcileRequest returns a reconcile.Request to be used throughout the reconciliation cycle,
 // regardless of which resource has triggered it.
-func (r *ReconcileHyperConverged) resolveReconcileRequest(ctx context.Context, logger logr.Logger, originalRequest reconcile.Request) (reconcile.Request, bool, error) {
+func (r *ReconcileHyperConverged) resolveReconcileRequest(ctx context.Context, logger logr.Logger, originalRequest reconcile.Request) (reconcile.Request, TriggeredBy, error) {
 
 	hcoTriggered, err := isTriggeredByHyperConverged(originalRequest)
 	if err != nil {
-		return reconcile.Request{}, hcoTriggered, err
+		return reconcile.Request{}, Other, err
 	}
 	if hcoTriggered {
 		logger.Info("Reconciling HyperConverged operator")
-		return originalRequest, hcoTriggered, nil
+		return originalRequest, Hco, nil
 	}
 
 	hc, err := getHyperConvergedNamespacedName()
 	if err != nil {
-		return reconcile.Request{}, hcoTriggered, err
+		return reconcile.Request{}, Other, err
 	}
 	resolvedRequest := reconcile.Request{
 		NamespacedName: hc,
 	}
 
 	apiServerCRTriggered, err := isTriggeredByApiServerCR(originalRequest)
 	if err != nil {
-		return reconcile.Request{}, hcoTriggered, err
+		return reconcile.Request{}, Other, err
 	}
 	if apiServerCRTriggered {
 		logger.Info("Triggered by ApiServer CR, refreshing it")
 		err = hcoutil.GetClusterInfo().RefreshAPIServerCR(ctx, r.client)
 		if err != nil {
-			return reconcile.Request{}, hcoTriggered, err
+			if hcoTriggered {
+				return reconcile.Request{}, Hco, err
+			}
+			return reconcile.Request{}, Other, err
```

Comment:
This case is not reachable, because we already check for it above, and exciting. Please drop this `if` sentence.